content/scripts/build.js

```diff
@@ -429,72 +437,99 @@ class Builder {
-        this.getLocaleRootFolders(source, { allLocales: true }).forEach(
-          filepath => {
-            walker(filepath, (folder, files) => {
-              if (source.isStumptown) {
-                files
-                  .filter(n => n.endsWith(".json"))
-                  .forEach(filename => {
-                    const filepath = path.join(folder, filename);
-                    this.processStumptownFileTitle(source, filepath);
-                  });
-                return;
-              }
-              if (
-                files.includes("index.html") &&
-                files.includes("index.yaml")
-              ) {
-                this.processFolderTitle(source, folder);
-              }
-            });
-          }
-        );
-      });
-      fs.writeFileSync(
-        allTitlesJsonFilepath,
-        JSON.stringify(this.allTitles, null, 2)
-      );
-      let t1 = new Date();
-      this.logger.info(
-        chalk.green(
-          `Building list of all titles took ${ppMilliseconds(t1 - t0)}`
-        )
+    // If we're going to generate all titles, we need all popularities.
+    // Normally this gets run later in the build process, but in the
+    // case of a needing the titles first, make sure popularities are set.
+    this.ensurePopularities();
+
+    this.logger.info("Building a list of ALL titles and URIs...");
+    let t0 = new Date();
+    this.sources.entries().forEach(source => {
+      this.getLocaleRootFolders(source, { allLocales: true }).forEach(
+        filepath => {
+          walker(filepath, (folder, files) => {
+            if (source.isStumptown) {
+              files
+                .filter(n => n.endsWith(".json"))
+                .forEach(filename => {
+                  const filepath = path.join(folder, filename);
+                  this.processStumptownFileTitle(source, filepath);
+                });
+              return;
+            }
+            if (files.includes("index.html") && files.includes("index.yaml")) {
+              this.processFolderTitle(source, folder);
+            }
+          });
+        }
       );
-    }
-  }
+    });
 
-  ensureAllChildren() {
-    this.ensureAllTitles();
-    Object.entries(this.allTitles).forEach(([uri, data]) => {
-      if (data.parent) {
-        const parentMdnURL = buildMDNUrl(data.parent.locale, data.parent.slug);
-        if (this.allTitles[parentMdnURL]) {
-          if (!(parentMdnURL in this.allChildren)) {
-            this.allChildren[parentMdnURL] = [];
+    // Only after *all* titles have been processed can we iterate over the
+    // mapping and figure out all translations.
+    // What this does is that it sets `.translations=[{slug, locale}, ...]`
+    // on every title that is the "translation_of". Practically, that means
+    // that every 'en-US' document that has been translated, will have a list
+    // of other locales and slugs.
+    let countBrokenTranslationOfDocuments = 0;
+    Object.values(this.allTitles)
```

Comment:
I took *your* `ensureAllChildren` but instead of renaming it into `ensureAllTranslations` I just cut its meat right into here. Right before the `content/_all-titles.json` file gets dumped. 

The only difference is that yours used a `for (const [uri, data] of Object.entries(this.allTitles)) {` whereas mine uses `Object.values(this.allTitles)....`.